src/components/Image/ImageLoader.jsx

```diff
@@ -41,11 +50,19 @@ class ImageLoader extends React.Component {
       this.img.onload = resolve
       this.img.onerror = reject
       this.img.src = src
-      this.timeout = setTimeout(reject, TTL)
+      this.timeout = setTimeout(
+        () => reject(new Error('Loading image took too long')),
+        TTL
+      )
+    }).finally(() => {
+      clearTimeout(this.timeout)
+      this.img.onload = this.img.onerror = null
+      this.img.src = ''
+      this.img = null
```

Comment:
Cleaning up, whether the image did load or not. Prevents fetching the raw version even  though the thumbnail was fetched, among other things.